catalog/app/utils/ResourceCache.js

```diff
@@ -57,83 +56,71 @@ const Action = tagged([
   'CleanUp', // { time: Date }
 ])
 
-const keyFor = (resource, input) => [resource.id, I.fromJS(resource.key(input))]
+const keyFor = (resource, input) => [resource.id, resource.key(input)]
```

Comment:
i think this is where things have been broken: `resource.key` doesn't always return string, in fact, it usually returns some object passed as an input, and this object gets used as a key for the entry, so we're basically doing smth like `cache[{ some: input }] = entry`, which turns `cache` var into `{ '[object Object]': entry }` (check redux logs and inspect the state to make sure), so, it kinda works when there's no conflicts for every resource (only single item is fetched) or every key is a string, but the logic is broken.
solution: use some kind of proper map implementation which can use arbitrary objects as keys (js native `Map` is okayish, but it's mutable, and i'm not sure about its performance when using it as immutable -- creating a new map on every change)